docs/standard/commandline/syntax.md

```diff
@@ -0,0 +1,577 @@
+* For the option result `*[ --fgcolor <White> ]`, the option wasn't specified on the command line, so the configured default was used. `White` is the effective value for this option.
+* `???-->` points to input that wasn't matched to any of the app's commands or options.
+
+### The `[suggest]` directive
+
+The `[suggest]` directive lets you search for commands when you don't know the exact command.
+
+```dotnetcli
+dotnet [suggest] buil
+```
+
+```output
+build
+build-server
+msbuild
+```
+
+## Design guidance
+
+The following sections present guidance that we recommend you follow when designing a CLI. Think of what your app expects on the command line as similar to what a REST API server expects in the URL. Consistent rules for REST APIs are what make them readily usable to client app developers. In the same way, users of your command-line apps will have a better experience if the CLI design follows common patterns.
+
+### Commands and subcommands
+
+If a command has subcommands, the command should function as an area, or a grouping identifier for the subcommands, rather than specify an action. When you invoke the app, you specify the grouping command and one of its subcommands. For example, try to run `dotnet tool`, and you get an error message because the `tool` command only identifies a group of tool-related subcommands, such as `install` and `list`. You can run `dotnet tool install`, but `dotnet tool` by itself would be incomplete.
+
+Some .NET CLI commands don't currently follow this recommended pattern. For example, `dotnet new` can do several different actions such as list, search, or install templates, but there are currently no subcommands under `new`.
+
+### Options as parameters
+
+Options should provide parameters to commands, rather than specifying actions themselves. This is a recommended design principle although it isn't always followed by `System.CommandLine` (`--help` displays help information) or the .NET CLI (`--list`, `--install`, and others function as commands under `dotnet new`).
+
+### Short-form aliases
+
+In general, we recommend that you minimize the number of short-form option aliases that you define.
+
+In particular, avoid using any of the following aliases differently than their common usage in the .NET CLI and other .NET command-line apps:
+
+* `-i` for `--interactive`.
+
+  This option signals to the user that they may be prompted for inputs to questions that the command needs answered. For example, prompting for a username.
+
+* `-o` for `--output`.
+
+  Some commands produce files as the result of their execution. This option should be used to help determine where those files should be located. In cases where a single file is created, this option should be a file path. In cases where many files are created, this option should be a directory path.
```

Comment:
@jonsequitur If there is a single option, is it reasonable to accept a directory? Do we want to explicitly suggest that the corresponding types be used, maybe via an example